torch/csrc/jit/passes/inliner.cpp

```diff
@@ -10,6 +10,51 @@ namespace prim {
 using namespace ::c10::prim;
 }
 
+namespace {
+void pushScopeInfo(Node* node) {
+  auto callstack_ptr = *(node->callstack());
+  const auto& vec = callstack_ptr->vec_with_module_info();
+  ScopePtr sc = c10::make_intrusive<Scope>();
+  for (const auto& tup : vec) {
+    const auto opt_module_instance_info = std::get<2>(tup);
+    if (opt_module_instance_info) {
+      const auto& module_instance_info = opt_module_instance_info.value();
+      if(module_instance_info.class_type()) {
+        const auto& class_type = module_instance_info.class_type();
+        const auto& instance_name = module_instance_info.instance_name();
+        auto type_name = class_type->name()->qualifiedName();
+        type_name = type_name.substr(type_name.find_last_of(".")+1);
+        std::string module_fn_name = type_name + "(" + instance_name + ")::";
+        module_fn_name += std::get<0>(tup)->name();
+        sc = sc->push(Symbol::scope(module_fn_name));
+      } else {
+        sc = sc->push(Symbol::scope("TYPE_INFO_UNKNOWN::"));
+      }
+    }
+    else {
+      std::string free_fn = "FreeFunction::" + std::get<0>(tup)->name();
+      sc = sc->push(Symbol::scope(free_fn));
+    }
+  }
+  node->setScope(sc);
+}
+
+void ReconstructScopeFromInlinedCallStackHelper(Block* block) {
```

Comment:
Do we really need this pass (why can't we just work directly with `vec_with_module_info` where we need the full attribute path)? If yes, then it should be in a separate PR.

Reply:
I mostly followed your pattern from reconstruct scopes, to actually update scope in a separate pass. Additionally it also helps to make testing easier. I can just run FileCheck on the graph.
It is possible that just call `ReconstructScopeFromInlinedCallStackHelper` directly after inlining, so you don't need to call this pass separately.
With respect to directly working with `vec_with_module_info`: That is certainly possible, but testing might be more complicated. How did you test `InlinedCallStack` changes?

Reply:
Also I can split this up in a separate PR, but I dont know a clean way to test the base PR then.

Reply:
>  How did you test InlinedCallStack changes?

Here are the tests for `InlinedCallStack`: https://github.com/pytorch/pytorch/blob/master/test/cpp/jit/test_misc.cpp#L1711-L1785
Probably, we could do a similar testing for this new functionality as well. I think it also would be a more rigorous testing as we wouldn't have a "man-in-the-middle" pass in the testing workflow.

I think I'm a bit reluctant with regard of this pass also because in a long term we were planning on getting rid of scopes entirely and replacing them with the InlinedCallStack, while this pass adds a new user of scopes. I don't think anyone is actively working in that direction, but it feels like a step in a slightly wrong direction.

Reply:
I am ok with this. We can reconstruct the same thing with a helper function elsewhere during bytecode export. I will make the changes.